parsers/EIA.py

```diff
@@ -70,54 +143,15 @@ def _fetch_production_or_consumption(zone_key, series_id, session=None,
     } for datapoint in raw_data['series'][0]['data']]
 
 
-def fetch_exchange(zone_key1, zone_key2, session=None, target_datetime=None, logger=None):
-    """Requests the last known power exchange (in MW) between two zones
-    Arguments:
-    zone_key1           -- the first country code
-    zone_key2           -- the second country code; order of the two codes in params doesn't matter
-    session (optional)      -- request session passed in order to re-use an existing session
-    target_datetime (optional)      -- string in form YYYYMMDDTHHZ
-    Return:
-    A list of dictionaries in the form:
-    {
-      'sortedZoneKeys': 'DK->NO',
-      'datetime': '2017-01-01T00:00:00Z',
-      'netFlow': 0.0,
-      'source': 'mysource.com'
-    }
-    where net flow is from DK into NO
-    """
-
-    sortedcodes = '->'.join(sorted([zone_key1, zone_key2]))
-
-    series_id = EXCHANGES[sortedcodes]
-    s = session or requests.Session()
-    exchange_series = Series(series_id=series_id, session=s)
-
-    if target_datetime:
-        raw_data = exchange_series.last_from(24, end=target_datetime)
-    else:
-        # Get the last 24 hours available.
-        raw_data = exchange_series.last(24)['series'][0]['data']
-
-    data = []
-    for datapoint in raw_data:
-        if sortedcodes == 'MX-BC->US-CA':
-            datapoint[1] = -1*datapoint[1]
-
-        exchange = {'sortedZoneKeys': sortedcodes,
-                    'datetime': parser.parse(datapoint[0]),
-                    'netFlow': datapoint[1],
-                    'source': 'mysource.com'}
-
-        data.append(exchange)
-
-    return data
+def main():
```

Comment:
Is there any particular reason to change this from how we usually do it?

Reply:
do you mean switching `fetch_exchange()` to reuse `_fetch_series()`? just unifying duplicate code.

if you mean `main()`, it keeps the relative import working when i run just this module, to test. specifically, `python parsers/EIA.py` fails on `from .ENTSEO import ...`. `python -c 'from parsers import EIA; EIA.main()'` works, but needs a top-level `main()` function to run. 😁 i'm happy to use an alternative if you know of one!

Reply:
Nope it's fine, I just wanted clarification.